ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -795,18 +867,40 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			// is not set, we will disable random Pod deletion by default.
 			if !enableInTreeAutoscaling || enableRandomPodDelete {
 				// diff < 0 means that we need to delete some Pods to meet the desired number of replicas.
-				randomlyRemovedWorkers := -diff
-				r.Log.Info("reconcilePods", "Number workers to delete randomly", randomlyRemovedWorkers, "Worker group", worker.GroupName)
-				for i := 0; i < int(randomlyRemovedWorkers); i++ {
-					randomPodToDelete := runningPods.Items[i]
-					r.Log.Info("Randomly deleting Pod", "progress", fmt.Sprintf("%d / %d", i+1, randomlyRemovedWorkers), "with name", randomPodToDelete.Name)
-					if err := r.Delete(ctx, &randomPodToDelete); err != nil {
-						if !errors.IsNotFound(err) {
-							return err
+				randomlyRemovedReplicas := -diff
```

Comment:
I am considering whether it is worth handling random deletions for multi-host PodSlices or not. Random deletion is a poor mechanism if it results in the removal of Pods with running workloads, which is worse than not deleting anything at all.

Reply:
Done.